monkey/monkey_island/readme.md

```diff
@@ -0,0 +1,103 @@
+
+5. Place portable version of OpenSSL
+    - Download from: <https://indy.fulgan.com/SSL/Archive/openssl-1.0.2l-i386-win32.zip>
+    - Extract content from bin folder to monkey_island\bin\openssl
+6. Download and install Microsoft Visual C++ redistributable for Visual Studio 2017
+    - Download and install from: <https://go.microsoft.com/fwlink/?LinkId=746572>
+7. Generate SSL Certificate
+    - run create_certificate.bat when your current working directory is monkey_island
+8. Create the monkey_island\cc\binaries folder and put Infection Monkey binaries inside (binaries can be found in releases on github)
+    monkey-linux-64 - monkey binary for linux 64bit
+    monkey-linux-32 - monkey binary for linux 32bit
+    monkey-windows-32.exe - monkey binary for windows 32bit
+    monkey-windows-64.exe - monkey binary for windows 64bit
+9. Install npm
+    - Download and install from: <https://www.npmjs.com/get-npm>
+10. Build Monkey Island frontend
+    - cd to 'monkey_island\cc\ui'
+    - run 'npm update'
+    - run 'npm run dist'
+
+#### How to run
+
+1. When your current working directory is monkey_island, run monkey_island\windows\run_server.bat
+
+### On Linux
+
+1. Get python 3.7 and pip if your linux distribution doesn't have it built in (following steps are for Ubuntu 16):
+    - `sudo add-apt-repository ppa:deadsnakes/ppa`
+    - `sudo apt-get update`
+    - `sudo apt install python3.7`
+    - `sudo apt install python3-pip`
+    - `python3.7 -m pip install pip`
+    - `sudo apt-get install python3.7-dev`
+2. Install required packages:
+    - `sudo apt-get install libffi-dev upx libssl-dev libc++1 openssl`
+3. Create the following directories in monkey island folder (execute from ./monkey):
+    - `mkdir -p ./monkey_island/bin/mongodb`
+    - `mkdir -p ./monkey_island/db`
+    - `mkdir -p ./monkey_island/cc/binaries`
+
+4. Install the packages from monkey_island/requirements.txt:
+    - `sudo python3.7 -m pip install -r ./monkey_island/requirements.txt`
+
+5. Put monkey binaries in /monkey_island/cc/binaries (binaries can be found in releases on github)
+    monkey-linux-64 - monkey binary for linux 64bit
+    monkey-linux-32 - monkey binary for linux 32bit
+    monkey-windows-32.exe - monkey binary for windows 32bit
+    monkey-windows-64.exe - monkey binary for windows 64bi
+
+6. Setup MongoDB (Use one of the two following options):
+    - Download MongoDB and extract it to /var/monkey_island/bin/mongodb:
+        1. Run `./monkey_island/linux/install_mongo.sh ./monkey_island/bin/mongodb`. This will download and extract the relevant mongoDB for your OS.
+
+    OR
+    - Use already running instance of mongodb
+        1. Run `set MONKEY_MONGO_URL="mongodb://<SERVER ADDR>:27017/monkeyisland"`. Replace '<SERVER ADDR>' with address of mongo server
+
+7. Generate SSL Certificate:
+    - `cd ./monkey_island`
+    - `./linux/create_certificate.sh`
+
+8. Install npm and node by running:
```

Comment:
I'd rather we link to official NPM installation instructions
https://github.com/nodesource/distributions/blob/master/README.md#deb

Reply:
@ShayNehmad are we pinned to a specific version of Node/NPM?

Reply:
We are not. These commands are taken from official NPM installation instructions, just saves the extra steps